crest/Assets/Crest/Crest/Scripts/Underwater/UnderwaterRenderer.Mask.cs

```diff
@@ -0,0 +1,131 @@
+            _oceanMaskCommandBuffer.SetGlobalTexture(sp_CrestOceanMaskTexture, _textureMask.colorBuffer);
+            _oceanMaskCommandBuffer.SetGlobalTexture(sp_CrestOceanMaskDepthTexture, _depthBuffer.depthBuffer);
+
+            PopulateOceanMask(
+                _oceanMaskCommandBuffer,
+                _camera,
+                OceanRenderer.Instance.Tiles,
+                _cameraFrustumPlanes,
+                _oceanMaskMaterial.material,
+                debug.disableOceanMask
+            );
+        }
+
+        internal static void InitialiseMaskTextures(RenderTextureDescriptor desc, ref RenderTexture textureMask, ref RenderTexture depthBuffer)
+        {
+            // Note: we pass-through pixel dimensions explicitly as we have to handle this slightly differently in HDRP
+            if (textureMask == null || textureMask.width != desc.width || textureMask.height != desc.height)
+            {
+                // @Performance: We should consider either a temporary RT or use an RTHandle if appropriate
+                // RenderTexture is a "native engine object". We have to release it to avoid memory leaks.
+                if (textureMask != null)
+                {
+                    textureMask.Release();
+                    depthBuffer.Release();
+                }
+
+                textureMask = new RenderTexture(desc);
+                textureMask.depth = 0;
+                textureMask.name = "Ocean Mask";
+                // @Memory: We could investigate making this an 8-bit texture instead to reduce GPU memory usage.
+                // We could also potentially try a half res mask as the mensicus could mask res issues.
+                textureMask.format = RenderTextureFormat.RHalf;
+                textureMask.Create();
+
+                depthBuffer = new RenderTexture(desc);
+                depthBuffer.depth = 24;
+                depthBuffer.enableRandomWrite = false;
+                depthBuffer.name = "Ocean Mask Depth";
+                depthBuffer.format = RenderTextureFormat.Depth;
+                depthBuffer.Create();
+            }
+        }
+
+        // Populates a screen space mask which will inform the underwater postprocess. As a future optimisation we may
+        // be able to avoid this pass completely if we can reuse the camera depth after transparents are rendered.
+        internal static void PopulateOceanMask(
+            CommandBuffer commandBuffer,
+            Camera camera,
+            List<OceanChunkRenderer> chunksToRender,
+            Plane[] frustumPlanes,
+            Material oceanMaskMaterial,
+            bool debugDisableOceanMask
+        )
+        {
+            GeometryUtility.CalculateFrustumPlanes(camera, frustumPlanes);
+
+            // Get all ocean chunks and render them using cmd buffer, but with mask shader.
+            if (!debugDisableOceanMask)
+            {
+                // Spends approx 0.2-0.3ms here on dell laptop
```

Comment:
potential candidate for removing.

otherwise maybe specialising it to be concrete: "2018 Dell XPS 15"

Reply:
Thanks. I've updated with the specific device.